pkg/client/dtclient/client.go

```diff
@@ -84,12 +84,21 @@ type ConfigClient interface {
 
 // DownloadSettingsObject is the response type for the ListSettings operation
 type DownloadSettingsObject struct {
-	ExternalId    string          `json:"externalId"`
-	SchemaVersion string          `json:"schemaVersion"`
-	SchemaId      string          `json:"schemaId"`
-	ObjectId      string          `json:"objectId"`
-	Scope         string          `json:"scope"`
-	Value         json.RawMessage `json:"value"`
+	ExternalId       string                    `json:"externalId"`
+	SchemaVersion    string                    `json:"schemaVersion"`
+	SchemaId         string                    `json:"schemaId"`
+	ObjectId         string                    `json:"objectId"`
+	Scope            string                    `json:"scope"`
+	Value            json.RawMessage           `json:"value"`
+	ModificationInfo *SettingsModificationInfo `json:"modificationInfo"`
+}
+
+type SettingsModificationInfo struct {
+	Deletable          bool          `json:"deletable"`
+	Modifiable         bool          `json:"modifiable"`
+	Movable            bool          `json:"movable"`
+	ModifiablePaths    []interface{} `json:"modifiablePaths"`
```

Comment:
I'd argue we don't need most of the fields here, even in near the future.

Reply:
Agreed, I wasn't sure about it either.

I've kept it in from the 'as-a-lib' perspective. 
I figure it has no real drawbacks to keep and doesn't hurt for the client to be aware of the full object.